frontend/apps/web/components/onboarding-checklist/OnboardingChecklist.tsx

```diff
@@ -0,0 +1,258 @@
+  PopoverContent,
+  PopoverTrigger,
+} from '@/components/ui/popover';
+import { Progress } from '@/components/ui/progress';
+import { useGetAccountOnboardingConfig } from '@/libs/hooks/useGetAccountOnboardingConfig';
+import {
+  AccountOnboardingConfig,
+  SetAccountOnboardingConfigRequest,
+  SetAccountOnboardingConfigResponse,
+} from '@neosync/sdk';
+import {
+  ArrowRightIcon,
+  ChevronDownIcon,
+  CircleIcon,
+  RocketIcon,
+} from '@radix-ui/react-icons';
+import { useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import { FaCheckCircle } from 'react-icons/fa';
+import { useAccount } from '../providers/account-provider';
+import { Button } from '../ui/button';
+import { Separator } from '../ui/separator';
+import { Skeleton } from '../ui/skeleton';
+import { toast } from '../ui/use-toast';
+
+interface OnboardingValues {
+  hasCreatedSourceConnection: boolean;
+  hasCreatedDestinationConnection: boolean;
+  hasCreatedJob: boolean;
+  hasInvitedMembers?: boolean;
+}
+
+interface Step {
+  id: string;
+  title: string;
+  href: string;
+  complete: boolean;
+}
+
+// create a map of {step:metadata} so we can construct that we need to we show in UI
+const stepsMap = {
+  hasCreatedSourceConnection: {
+    title: 'Create a source connection',
+    href: '/connection',
+  },
+  hasCreatedDestinationConnection: {
+    title: 'Create a destination connection',
+    href: '/connection',
+  },
+  hasCreatedJob: { title: 'Create a Job', href: '/job' },
+  hasInvitedMembers: { title: 'Invite members', href: '/settings' },
+};
+
+export default function OnboardingChecklist() {
+  const { account } = useAccount();
+  const { data, isLoading, isValidating } = useGetAccountOnboardingConfig(
+    account?.id ?? ''
+  );
+  const [progress, setProgress] = useState<number>(0);
+  const [isOpen, setIsOpen] = useState(false);
+  const [values, setValues] = useState<Step[]>([]);
+  const [showGuide, setShowGuide] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (data?.config) {
+      // if account is a personal account remove the invite a team member step
+      const updatedConfig =
+        account?.type == 1
+          ? (({ hasInvitedMembers, ...rest }) => rest)(data.config)
+          : data.config;
+
+      // check if any of the steps are false, if so, then surface the guide
+      if (account?.type == 1 && !didUserCompleteOnboarding(updatedConfig)) {
+        setShowGuide(true);
+        setIsOpen(true);
+      }
+
+      // construct the stepsArray so that we can map over it.
+      const stepsArray = Object.entries(updatedConfig).map(([key, value]) => {
+        return {
+          id: key,
+          title: stepsMap[key as keyof typeof stepsMap].title,
+          href: stepsMap[key as keyof typeof stepsMap].href,
+          complete: value,
+        };
+      });
+      setValues(stepsArray);
+
+      // calculate the progress percentage
+      const progressPercentage =
+        account?.type == 1
```

Comment:
we have named enums for this to make it a little more clear what the account type is. you can see how were doing that on the settings page